### PR TITLE
Set seed at top level for tests that random before run_tests call

### PR DESCRIPTION
**Pull request by ssnl**
Fixes some intermittent test failures due to initialization not respecting SEED option.

cc: @houseroad

### CODE REVIEW

File: test/common.py [hunk 1]
```diff
@@ -21,31 +21,31 @@
 SEED = 0
 SEED_SET = 0
 ACCEPT = False
+UNITTEST_ARGS = None
 
 
-# TODO rename me
-def parse_set_seed_once():
+def parse_args_set_seed_if_needed():
```
Comment (apaszke) [inline test/common.py]: Can't we just always run this in `common.py` instead of adding it in every file separately?

Comment (SsnL) [reply test/common.py]: Good point!

**Comment by SsnL** [comment]
Very interesting test failure. The `TestNN.test_L1Loss` fails if its `target=torch.randn(2, 3, 4)` is generated after setting seed (https://github.com/pytorch/pytorch/blob/master/test/common_nn.py#L239). I'm investigating this.

**Comment by ezyang** [comment]
I have observed some tests are sensitive to initial seed. So it's worth seeing if you're getting unlucky with seed selection or not.

**Comment by SsnL** [comment]
I don't think that this is a seed issue. I tested a couple different seeds and the test fails on all of them. Interestingly, `TestNN.test_L1Loss_cuda` doesn't fail.

**Comment by SsnL** [comment]
Yeah, I'm almost certain that something is wrong with `seed` and the cpu kernel. 

1. Save unseeded `target` and load after seeding. Test passes.
2. Save seeded `target` and load before seeding. Test fails.

**Comment by apaszke** [comment]
Note that CPU tests do a lot more checks than CUDA. `_cuda` tests are pretty much like run CPU, run GPU, run CPU grad, run GPU grad, compare everything, done. So if CUDA test passes it means that CPU and GPU give consistent outputs

**Comment by SsnL** [comment]
I'll close this PR and re-open another one for clarity.

**Pull request closed without merging**